academic_observatory_workflows/workflows/pubmed_telescope.py

```diff
@@ -0,0 +1,1345 @@
+                                )
+
+                                # Remove the old record from merged_data
+                                del merged_dict[key]
+
+                                update_counter += 1
+
+                        # Convert back to a list.
+                        merged_data = list(merged_dict.values())
+
+                    # Add on the newer records.
+                    # This will also merge all of the deletion records into one file so that can be removed from the main table.
+                    merged_data.extend(incoming_data)
+
+                logging.info(f"Found the following number of records that were updated - {update_counter}")
+
+                merged_output_file = changefile.merged_transform_file_path(
+                    entity.type, first_file_index, changefile.file_index
+                )
+
+                merged_updatefiles[entity.name] = merged_output_file
+
+                logging.info(f"Writing to file - {merged_output_file}")
+
+                # Write to compressed *.jsonl.gz
+                with gzip.open(merged_output_file, "w") as f_out:
+                    for line in merged_data:
+                        f_out.write(str.encode(json.dumps(line, cls=PubMedCustomEncoder) + "\n"))
+
+        # Push list of merged transformed files to Xcom for upload step.
+        ti: TaskInstance = kwargs["ti"]
+        ti.xcom_push(key="merged_updatefiles", value=merged_updatefiles)
+
+    def upload_transformed(self, release: PubMedRelease, **kwargs):
+        """
+        Upload the merged transformed files to GCS.
+        """
+
+        ti: TaskInstance = kwargs["ti"]
+        merged_updatefiles = ti.xcom_pull(key="merged_updatefiles")
+        files_to_upload = list(merged_updatefiles.values())
+
+        logging.info(f"files_to_upload - {files_to_upload}")
+
+        success = gcs_upload_files(
+            bucket_name=self.cloud_workspace.transform_bucket,
+            file_paths=files_to_upload,
+        )
+
+        if not success:
+            raise AirflowException(
+                f"Unable to upload downloaded changefiles to GCS bucket {self.cloud_workspace.download_bucket}"
+            )
+
+    def bq_ingest_update_tables(self, release: PubMedRelease, **kwargs):
+        """
+        Ingest data into tables for this particular release.
+
+        If first run of the workflow, it will only trasnfer the baseline table in BQ, otherwise it will
+        ingest the addition and deletion tables as shards.
+        """
+
+        ti: TaskInstance = kwargs["ti"]
+        merged_updatefiles = ti.xcom_pull(key="merged_updatefiles")
+
+        # If its the first run of the workflow, only upload the baseline table to the main table.
+        if release.is_first_run:
+            # Create the dataset if it doesn't exist already.
+            try:
+                bq_create_dataset(
```

Comment:
You don't need the try/except, bq_create_dataset will catch the error raised for an existing dataset